src/jobs/AbstractJob.py

```diff
@@ -0,0 +1,12 @@
+import abc
+from typing import List
+
+
+class AbstractJob(abc.ABC):
```

Comment:
I figure there are probably some argument that the jobs will have in common (api and api configuration), and some arguments that are job specific (number of iterations/samples etc). I wonder if it is possible to architect it so that JobRunner.py supplies some arguments and passes an argparse.ArgumentParser object to the jobs. Then the JobRunner.py can do some of the common setup (eg with Apis and such) (Look at WikiRacer.py to see how to initialize the apis), and pass the api and the arguments object to the run method of the job.

Thoughts?

Reply:
I don't think it's trivial to do that, since argparse isn't really designed to partially parse the argv. I think it's more work than it's worth since we won't have that many one off jobs.

Reply:
Fair enough, although I think it could probably be done (based on the usage of argparse)